x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1641,6 +1645,40 @@ public boolean test(String profile, InetSocketAddress peerAddress) {
                 sslConfiguration = null;
                 populateClientCertificate = (channel, threadContext) -> {};
             }
+            final AuthenticationService authenticationService = this.authcService.get();
+            final ThreadContext threadContext = this.threadContext.get();
+            final TriConsumer<HttpPreRequest, Channel, ActionListener<ValidationResultContext>> authenticateMessage = (
+                httpRequest,
+                channel,
+                listener) -> {
+                // step 1: stash the current thread's context and wrap the authentication result listener for context-preserving because
```

Comment:
I've pushed [2586e17](https://github.com/elastic/elasticsearch/pull/95112/commits/2586e1776a8af13f40f141856aada7f8ea8f2769) .
I tried to not go into details about what the thread context should contain, because that might change without warning (the purpose of `stashContext` is to ignore whatever the context contains before it).